@compnerd I didn't know that `__attribute__ ((warn_unused_result))` is not supported on Windows, should I make it a macro or is there a better option?

```
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Syntax/Syntax.h(505): warning C4996: 'swift::syntax::RawSyntax::dump': only for use in the debugger
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Sema/ConstraintSystem.h(4591): error C2059: syntax error: '('
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Sema/ConstraintSystem.h(4591): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Sema/ConstraintSystem.h(4591): error C2059: syntax error: ')'
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Sema/ConstraintSystem.h(4591): error C2059: syntax error: ')'
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Sema/ConstraintSystem.h(4592): warning C4183: '__attribute__': missing return type; assumed to be a member function returning 'int'

```

--- a/include/swift/Sema/ConstraintSystem.h
+++ b/include/swift/Sema/ConstraintSystem.h
@@ -4587,6 +4587,7 @@ public:
   /// Generate constraints for the given solution target.
   ///
   /// \returns true if an error occurred, false otherwise.
+  LLVM_NODISCARD
   bool generateConstraints(SolutionApplicationTarget &target,
                            FreeTypeVariableBinding allowFreeTypeVariables);
 
@@ -4595,11 +4596,13 @@ public:
   /// \param closure the closure expression
   ///
   /// \returns \c true if constraint generation failed, \c false otherwise
+  LLVM_NODISCARD
   bool generateConstraints(ClosureExpr *closure);
 
   /// Generate constraints for the given (unchecked) expression.
   ///
   /// \returns a possibly-sanitized expression, or null if an error occurred.
+  LLVM_NODISCARD
   Expr *generateConstraints(Expr *E, DeclContext *dc,
                             bool isInputExpression = true);
 
@@ -4607,6 +4610,7 @@ public:
   /// value of the given expression.
   ///
   /// \returns a possibly-sanitized initializer, or null if an error occurred.
+  LLVM_NODISCARD
   Type generateConstraints(Pattern *P, ConstraintLocatorBuilder locator,
                            bool bindPatternVarsOneWay,
                            PatternBindingDecl *patternBinding,
@@ -4616,6 +4620,7 @@ public:
   ///
   /// \returns true if there was an error in constraint generation, false
   /// if generation succeeded.
+  LLVM_NODISCARD
   bool generateConstraints(StmtCondition condition, DeclContext *dc);
 
   /// Generate constraints for a case statement.
@@ -4625,6 +4630,7 @@ public:
   ///
   /// \returns true if there was an error in constraint generation, false
   /// if generation succeeded.
+  LLVM_NODISCARD
   bool generateConstraints(CaseStmt *caseStmt, DeclContext *dc,
                            Type subjectType, ConstraintLocator *locator);
 
@@ -4668,6 +4674,7 @@ public:
   /// \param propertyType The type of the wrapped property.
   ///
   /// \returns true if there is an error.
+  LLVM_NODISCARD
   bool generateWrappedPropertyTypeConstraints(VarDecl *wrappedVar,
                                               Type initializerType,
                                               Type propertyType);

--- a/lib/Sema/CSSimplify.cpp
+++ b/lib/Sema/CSSimplify.cpp
@@ -9914,8 +9914,9 @@ bool ConstraintSystem::resolveClosure(TypeVariableType *typeVar,
       }
 
       if (!paramDecl->getName().hasDollarPrefix()) {
-        generateWrappedPropertyTypeConstraints(paramDecl, backingType,
-                                               param.getParameterType());
+        if (generateWrappedPropertyTypeConstraints(paramDecl, backingType,
+                                                   param.getParameterType()))
+          return false;
       }
 
       auto result = applyPropertyWrapperToParameter(backingType, param.getParameterType(),

--- a/unittests/Sema/PlaceholderTypeInferenceTests.cpp
+++ b/unittests/Sema/PlaceholderTypeInferenceTests.cpp
@@ -36,7 +36,9 @@ TEST_F(SemaTest, TestPlaceholderInferenceForArrayLiteral) {
 
   ConstraintSystem cs(DC, ConstraintSystemOptions());
   cs.setContextualType(arrayExpr, {arrayRepr, arrayTy}, CTP_Initialization);
-  cs.generateConstraints(target, FreeTypeVariableBinding::Disallow);
+  auto failed = cs.generateConstraints(target, FreeTypeVariableBinding::Disallow);
+  ASSERT_FALSE(failed);
+
   SmallVector<Solution, 2> solutions;
   cs.solve(solutions);
 
@@ -75,7 +77,9 @@ TEST_F(SemaTest, TestPlaceholderInferenceForDictionaryLiteral) {
 
   ConstraintSystem cs(DC, ConstraintSystemOptions());
   cs.setContextualType(dictExpr, {dictRepr, dictTy}, CTP_Initialization);
-  cs.generateConstraints(target, FreeTypeVariableBinding::Disallow);
+  auto failed = cs.generateConstraints(target, FreeTypeVariableBinding::Disallow);
+  ASSERT_FALSE(failed);
+
   SmallVector<Solution, 2> solutions;
   cs.solve(solutions);
 

--- a/unittests/Sema/UnresolvedMemberLookupTests.cpp
+++ b/unittests/Sema/UnresolvedMemberLookupTests.cpp
@@ -33,9 +33,11 @@ TEST_F(SemaTest, TestLookupAlwaysLooksThroughOptionalBase) {
   auto *UMCRE = new (Context) UnresolvedMemberChainResultExpr(UME, UME);
 
   ConstraintSystem cs(DC, ConstraintSystemOptions());
-  cs.generateConstraints(UMCRE, DC);
+  auto *expr = cs.generateConstraints(UMCRE, DC);
+  ASSERT_TRUE(expr);
+
   cs.addConstraint(
-      ConstraintKind::Conversion, cs.getType(UMCRE), intOptType,
+      ConstraintKind::Conversion, cs.getType(expr), intOptType,
       cs.getConstraintLocator(UMCRE, ConstraintLocator::ContextualType));
   SmallVector<Solution, 2> solutions;
   cs.solve(solutions);
@@ -67,10 +69,12 @@ TEST_F(SemaTest, TestLookupPrefersResultsOnOptionalRatherThanBase) {
   auto *UMCRE = new (Context) UnresolvedMemberChainResultExpr(UME, UME);
 
   ConstraintSystem cs(DC, ConstraintSystemOptions());
-  cs.generateConstraints(UMCRE, DC);
+  auto *expr = cs.generateConstraints(UMCRE, DC);
+  ASSERT_TRUE(expr);
+
   cs.addConstraint(
-      ConstraintKind::Conversion, cs.getType(UMCRE), intOptType,
-      cs.getConstraintLocator(UMCRE, ConstraintLocator::ContextualType));
+      ConstraintKind::Conversion, cs.getType(expr), intOptType,
+      cs.getConstraintLocator(expr, ConstraintLocator::ContextualType));
   SmallVector<Solution, 2> solutions;
   cs.solve(solutions);
 

